scalyr_agent/sharded_copying_manager/copying_manager.py

```diff
@@ -1148,19 +1027,22 @@ def __create_log_processors_for_log_matchers(
             if path in self.__log_paths_being_processed:
                 pending_removal[path] = True
 
+
         # iterate over the log_matchers while we create the LogFileProcessors
         for matcher in log_matchers:
             for new_processor in matcher.find_matches(
                 self.__log_paths_being_processed,
                 checkpoints,
                 copy_at_index_zero=copy_at_index_zero,
+                get_log_processor_factory=self.__get_log_processor_factory,
             ):
-                self.__log_processors.append(new_processor)
-                self.__log_paths_being_processed[new_processor.log_path] = new_processor
```

Comment:
Now log processor is added to the worker inside the `matcher.find_matches` function.